### PR TITLE
Add DB migration support using goose (based on pressly fork)

### PR DESCRIPTION
**Pull request by przemyslaw-dobrowolski-cl**
This commit replaces migration mechanism built-in simple implementation with a complete solution based on goose, pressly fork.

### CODE REVIEW

**Comment by przemyslaw-dobrowolski-cl** [comment]
@p-kozlowski @Zimnx @mmatczuk Please review.

File: db/migration/migration.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+var logger = log.NewLogger()
+
+func LoadConfig(configFile string) (err error) {
+	config := util.GetConfig()
+
+	err = config.ReadConfig(configFile)
+
+	if err != nil {
+		fmt.Printf("error: failed to load config: %s\n", err.Error())
```
Comment (p-kozlowski) [inline db/migration/migration.go]: you've got a logger, so why Printf? (same elsewhere)

File: db/migration/migration.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+package migration
+
+import (
+	"fmt"
+
+	"database/sql"
+	"github.com/cloudwan/gohan/log"
+	"github.com/cloudwan/gohan/util"
+	"github.com/pressly/goose"
```
Comment (p-kozlowski) [inline db/migration/migration.go]: add goose to vendor (this is why circleci fails)

File: server/server.go [hunk 3]
```diff
@@ -252,7 +259,9 @@ func NewServer(configFile string) (*Server, error) {
 		return nil, fmt.Errorf("invalid sync type: %s", syncType)
 	}
 
-	server.connectDB()
+	if err = server.connectDB(); err != nil {
+		return nil, err
+	}
```
Comment (p-kozlowski) [inline server/server.go]: not a single test case for this new feature...

File: db/migration/migration.go (hunk 1 above)
Comment (przemyslaw-dobrowolski-cl) [inline db/migration/migration.go]: When running gohan in migration mode, we assume that we operate directly with gohan. All messages are put to stdout andwe do not have to setup logging and provide proper configuration files.

**Comment by nati** [comment]
Could you update a doc for this?

File: db/migration/migration.go (hunk 2 above)
Comment (przemyslaw-dobrowolski-cl) [inline db/migration/migration.go]: added

**Comment by przemyslaw-dobrowolski-cl** [comment]
@nati sure, docs added

**Pull request merged**